### PR TITLE
Support for EKS Connector

### PR DESCRIPTION
**Pull request by cPu1**
### Description

Adds support for EKS Connector. 

# Registering non-EKS clusters with EKS Connector
The EKS Console provides a single pane of glass to manage all your Kubernetes clusters, including those hosted on
other cloud providers, via EKS Connector. This process requires registering the cluster with EKS and running the 
EKS Connector agent on the external Kubernetes cluster. 

`eksctl` simplifies registering non-EKS clusters by creating the required AWS resources and generating Kubernetes manifests 
for EKS Connector to apply to the external cluster.

## Register Cluster
To register or connect a non-EKS Kubernetes cluster, run
 
```shell
$ eksctl register cluster --name <name> --provider <provider>
2021-08-19 13:47:26 [ℹ]  creating IAM role "eksctl-20210819194112186040"
2021-08-19 13:47:26 [ℹ]  registered cluster "<name>" successfully
2021-08-19 13:47:26 [ℹ]  wrote file eks-connector.yaml to <current directory>
2021-08-19 13:47:26 [ℹ]  wrote file eks-connector-binding.yaml to <current directory>
2021-08-19 13:47:26 [!]  note: ClusterRoleBinding in "eks-connector-binding.yaml" gives cluster-admin permissions to IAM identity "<aws-arn>", edit if required; read https://eksct.io/usage/eks-connector for more info
2021-08-19 13:47:26 [ℹ]  run `kubectl apply -f eks-connector.yaml,eks-connector-binding.yaml` before <expiry> to connect the cluster

```

This command will register the cluster and write two files `eks-connector.yaml` and `eks-connector-binding.yaml` that contain
the Kubernetes manifests for EKS Connector that must be applied to the external cluster before the registration expires.

!!!note
`eks-connector-binding.yaml` contains a `ClusterRoleBinding` that gives `cluster-admin` permissions to the calling
IAM identity and must be edited accordingly if required before applying it to the cluster.

To provide an existing IAM role to use for EKS Connector, pass it via `--role-arn` as in: 

```shell
$ eksctl register cluster --name <name> --provider <provider> --role-arn=<role-arn>
```

If the cluster already exists, eksctl will return an error.

## Deregister cluster

To deregister or disconnect a registered cluster, run

```shell
$ eksctl deregister cluster --name <name>
2021-08-19 16:04:09 [ℹ]  unregistered cluster "<name>" successfully
2021-08-19 16:04:09 [ℹ]  run `kubectl delete namespace eks-connector` and `kubectl delete -f eks-connector-binding.yaml` on your cluster to remove EKS Connector resources
```

This command will deregister the external cluster and remove its associated AWS resources, but you are required to remove the 
EKS connector Kubernetes resources from the cluster.

### Checklist
- [ ] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/connector/kubernetes.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	}
+
+	if err := writeFile("eks-connector.yaml", manifestList.ConnectorResources); err != nil {
+		return err
+	}
+
+	if err := writeFile("eks-connector-binding.yaml", manifestList.ClusterRoleResources); err != nil {
+		return err
+	}
+	// TODO blog link
```
Comment (cPu1) [inline pkg/connector/kubernetes.go:80]: I couldn't find a link. This isn't a blocker for releasing.

File: integration/tests/addons/addons_test.go [hunk 2]
```diff
@@ -12,8 +12,6 @@ import (
 	. "github.com/weaveworks/eksctl/integration/runner"
 	"github.com/weaveworks/eksctl/integration/tests"
 	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
-	"github.com/weaveworks/eksctl/pkg/eks"
-	kubewrapper "github.com/weaveworks/eksctl/pkg/kubernetes"
```
Comment (aclevername) [inline integration/tests/addons/addons_test.go:16]: this is causing the linter to fail, not sure why this happened :thinking:

**Review by aclevername — APPROVED** [review]

**Comment by aclevername** [comment]
can we run the new integration tests before merging? Not the whole suite as that will happen during release

**Comment by cPu1** [comment]
> can we run the new integration tests before merging? Not the whole suite as that will happen during release

Sure, I already ran them earlier today but I'm going to run it again before merging.

File: integration/tests/addons/addons_test.go (hunk 2 above)
Comment (cPu1) [reply integration/tests/addons/addons_test.go]: I think this may have happened when I rebased it. It's fixed now so the linter should pass.

**Comment by Callisto13** [comment]
@nikimanoledaki are you done running your destroy scripts on the int test account?

**Pull request merged**